extensions/socketio/src/socketio.sequence.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/socketio
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Context, ControllerClass, CoreBindings, inject} from '@loopback/core';
+import {SocketIoBindings} from './keys';
+import {
+  SocketIoInvokeMethod,
+  SocketIoRejectMethod,
+  SocketIoSendMethod,
+  SocketIoSequence,
+} from './types';
+
+export class DefaultSocketIoSequence implements SocketIoSequence {
+  constructor(
+    @inject.context() protected context: Context,
+    @inject(CoreBindings.CONTROLLER_CURRENT)
+    protected controller: ControllerClass,
+    @inject(SocketIoBindings.INVOKE_METHOD)
+    protected invoke: SocketIoInvokeMethod,
+    @inject(SocketIoBindings.SEND_METHOD)
+    protected send: SocketIoSendMethod,
+    @inject(SocketIoBindings.REJECT_METHOD)
+    protected reject: SocketIoRejectMethod,
+  ) {}
```

Comment:
We should introduce a middleware based sequence for more flexibility and composability.